packages/editor/api-report.md

```diff
@@ -883,9 +902,17 @@ export class Editor extends EventEmitter<TLEventMap> {
         isInViewport: boolean;
         maskedPageBounds: Box2d | undefined;
     }[];
-    reparentShapes(shapes: TLShape[], parentId: TLParentId, insertIndex?: string): this;
+    reparentShapes(shapes: TLShape[], parentId: TLParentId, opts?: {
+        insertIndex?: string;
+        ephemeral?: boolean;
+        squashing?: boolean;
+    }): this;
     // (undocumented)
-    reparentShapes(ids: TLShapeId[], parentId: TLParentId, insertIndex?: string): this;
+    reparentShapes(ids: TLShapeId[], parentId: TLParentId, opts?: {
+        insertIndex?: string;
+        ephemeral?: boolean;
+        squashing?: boolean;
+    }): this;
```

Comment:
You'll see a LOT of overloaded functions in this PR. Basically instead of having a `doSomething` and `doSomethingById`, we collapse the APIs into one overloaded function. I always stayed away from this for perf worries, but the perf difference is insignificant.